include/xal.h

```diff
@@ -44,4 +44,5 @@ struct xal {
 	struct xal_inode *root;	 ///< Root of the file-system
 	struct xal_sb sb;
 	uint8_t be[XAL_BACKEND_SIZE];
+	bool dirty;              ///< Whether the file system has changed since last index
```

Comment:
If you declare this ``atomic_bool``, and include ``<stdatomic.h>``, then you can use ``atomic_store()``/ ``atomic_load()`` for sync. among threads.